packages/drivers/odsp-driver/src/snapshotRepresentation.ts

```diff
@@ -0,0 +1,486 @@
+}
+
+/**
+ * This contains mapping of Marker Codes to number of bytes in which the corresponding data
+ * will be stored.
+*/
+const codeToBytesMap = {
+    // Integer code to bytes
+    1: 0,
+    3: 1,
+    5: 2,
+    7: 4,
+    9: 8,
+
+    // String code to Bytes
+    13: 0,
+    14: 1,
+    15: 2,
+    16: 4,
+
+    // Binary code to bytes
+    32: 0,
+    33: 1,
+    34: 2,
+    35: 4,
+    36: 8,
+};
+
+/**
+ * This contains mapping of number of bytes to Marker Codes representing the corresponding Integer.
+*/
+const integerBytesToCodeMap = {
+    0: 1,
+    1: 3,
+    2: 5,
+    4: 7,
+    8: 9,
+};
+
+/**
+ * This contains mapping of number of bytes representing the corresponding string length to Marker Codes.
+*/
+const stringBytesToCodeMap = {
+    0: 13,
+    1: 14,
+    2: 15,
+    4: 16,
+};
+
+/**
+ * This contains mapping of number of bytes representing the corresponding length in which actual data
+ * will be stored to Marker Codes.
+*/
+const binaryBytesToCodeMap = {
+    0: 32,
+    1: 33,
+    2: 34,
+    4: 35,
+    8: 16,
+};
+
+/**
+ * Calculate how many bytes are required to encode an integer. This is always multiple of 2.
+ * So if 6 bytes are required to store an integer than it will return 8.
+ * @param num - number to encode.
+ */
+function calcLength(numArg: number) {
+    let num = numArg;
+    let lengthLen = 0;
+    while (num > 0) {
+        num = Math.floor(num / 256);
+        lengthLen++;
+    }
+    let res = 0;
+    let index = 0;
+    while (res < lengthLen) {
+        res = Math.pow(2, index);
+        index++;
+    }
+    return res;
+}
+
+export function iteratePairs<T>(it: IterableIterator<T>) {
+    const res: IterableIterator<[T, T]> = {
+        next: () => {
+            const a = it.next();
+            if (a.done) {
+                return { value: undefined, done: true };
+            }
+            const b = it.next();
+            assert(b.done !== true, "Should be a pair");
+            return { value: [a.value, b.value], done: b.done };
+        },
+        [Symbol.iterator]: () => { return res; },
+    };
+    return res;
+}
+
+/**
+ * Helper function that returns iterator from an object
+ * @param obj - object that supports iteration
+ */
+export function iterate<T>(obj: {[Symbol.iterator]: () => IterableIterator<T>}) {
+    return obj[Symbol.iterator]();
+}
+
+/**
+ * Base class to represent binary blob element.
+ * Binary blob is one of three types supported as a leaf node of a tree.
+ * Note: concrete implementations (derived classes) are not exposed from this module
+ */
+export abstract class BlobCore {
+    public abstract get buffer(): Uint8Array;
+    /**
+     * Represents a blob.
+     * @param useStringMarkerCode - Represents is the string marker code should be used when representing or
+     * the byte marker code.
+     */
+    constructor(private readonly useStringMarkerCode: boolean = false) {}
+
+    public toString(encoding = stringEncoding) {
```

Comment:
So in my original exploration I chose to support any encoding with utf-8. But I'm not sure what's the story with encodings in ZipIt format. Does it have same lazy approach to encoding? Or it mandates specific encoding? If it's the later, then we should not allow callers to overwrite it.

Reply:
Removed as we only bothers about converting to utf8 string here. zipIt can encode utf8 or base64 but we only want utf8 here.